### ISSUE TITLE
Needs Manual

### ISSUE THREAD

**Issue by sunfish62** [jonasoreland/runnerup#565]
Hello. I am brand new to your app, and I think it will be a great app for runners and others, but the fact that there is no documentation renders the software largely unaccessible to most users. 

Without a manual, users who lack knowledge of exercise theory will not understand the meaning of settings such as the "Convert rest on ..." settings (under Workout).

Without a manual, users have no way of knowing how to use the Advanced or Manual tabs to create a workout (if such even exists).

Without a manual, users have no way to create their own *.json files and load them in to RunnerUp for their own custom programs.

Without a manual, users have no way of knowing how to save changes to an existing workout (if such is even possible).

Searches here and on the web have not yielded any substantive resources (website, wiki, online manual, user forum, etc.) to assist the new user with your software.

So, how do you expect anyone new to use your app?

**Comment by gerhardol** [comment]
As is it is now, the code is the manual. The wiki is open for anyone to edit. Documentation can be boring. I have written a lot of documentation in in projects I control. just to realize that few are interested anyway, so documentation  is not really rewarding. Please add if you find anything worthwhile.

What I know, there is no way to add own .json files without changing the code.

**Comment by sunfish62** [comment]
"The code is the manual" only works for programmers. I am disappointed that such a promising app would take such a short-sighted perspective on growing its user base. As it stands right now, your audience is coders who run. 

So, should I flag the app as inappropriate on the Google Play site, since support is not available?

**Comment by Marx2** [comment]
Have you paid for this application, that you demand support? Move your ass and make documentation if you miss it, don't try to frighten developers with flagging

**Comment by sunfish62** [comment]
Ah. The old "you get what you pay for" excuse. Works every time. Without a doubt, that's the way to grow your user base.

I'll be happy to work on documentation.

**Comment by sunfish62** [comment]
So, having created an initial documentation framework for this on the wiki, please help me understand the meaning of the following Workout settings:

"Convert rest on Interval tab/Convert rest-step with type distance to recovery-step for Interval-tab"
"Convert rest on Advanced tab/Convert rest-step with type distance to recovery-step for Advanced-tab"
"Target pace moving average seconds"
"Target pace grace seconds"
"Autolap during intervals"
"Add countdown after step that ends w.." (this is how the setting appears on my device)

On a different point, on the Interval tab, there are check boxes for warm up and cool down; these items are greyed out on my device. Assuming that they allow a warm up and cool down for the session, what are they set to, and how are they changed?

**Comment by gerhardol** [comment]
RunnerUp is not my project. I just submit patches. 
A quick attempt to answer (I do not use the workout feature in RU myself.) I have browsed the code, but will not spend hours myself.
> Comments below

"Convert rest on Interval tab/Convert rest-step with type distance to recovery-step for Interval-tab"
"Convert rest on Advanced tab/Convert rest-step with type distance to recovery-step for Advanced-tab"
> If you setup Rest steps with distance, convert to recovery instead

"Target pace moving average seconds"
> When you have a Target Pace feedback setup, the averaging time

"Target pace grace seconds"
> When you have a Target Pace feedback setup, how much time outside the pace before reporting
  
"Autolap during intervals"
> If you have set Autolap, do you want it during the intervals?

"Add countdown after step that ends w.." (this is how the setting appears on my device)
> Add countdown after step that ends with user press
> Rotation is locked in Start to not confuse during workouts. Applies to preferences too.
 
On a different point, on the Interval tab, there are check boxes for warm up and cool down;
>The values are never used, I assume that warmup/cooldown is hardcoded here.

**Comment by sunfish62** [comment]
Thanks for responding. Your replies don't actually explain the features, though. Let me try again with a couple of specifics.

The Convert rest options are unclear to me. What precisely do the terms "rest-step" and "recovery-step" actually mean, and why would a user want to convert from one to another? As I originally said, I am not a sports person, but I run. Thus, I do not understand these terms or their significance.

WRT the moving average setting: I presume you mean this is number of seconds over which to average a pace. I also presume that the samples are per-second. As a practical matter, what kind of setting would someone want to use in this setting? 10 seconds? 15? 30?

What exactly is "Autolap"? 

You say the warmup/cooldown settings are hard-coded. What are they set to?

**Comment by gerhardol** [comment]
Some more attempts to describe structured intervals (there is nothing about programming in the response). There are likely better descriptions.
Again, I do not use this feature in RunnerUp and to a small extent with a standalone running watch.

> The Convert rest options are unclear to me. What precisely do the terms "rest-step" and "recovery-step" actually mean, and why would a user want to convert from one to another? As I originally said, I am not a sports person, but I run. Thus, I do not understand these terms or their significance.

A rest step is the rest part in an interval. Garmin (originally at least) only had Active and Rest steps. This is the basis in structured intervals, exactly what this feature is about. You can have variants of Rest like warmup/cooldown, inactive etc and of course have different synonyms for Active too. Some care to analyze the inactive steps separately from rest laps.
Personally, I only care about Active/Rest and cannot explain why 

> WRT the moving average setting: I presume you mean this is number of seconds over which to average a pace. I also presume that the samples are per-second. As a practical matter, what kind of setting would someone want to use in this setting? 10 seconds? 15? 30?

You can configure sampling time in RunnerUp.
The moving average depends to a large extent on the GPS quality and theterrain, there is no universal settings. You probably want to have over 20s at least, maybe up to a minute to get stable readings. 
This feature needs a refresh to work better with GPS (maybe supported by the accelerometer).

> What exactly is "Autolap"?

The watch creates laps at a configured time or distance. A typical setting is autolap every kilometer.

> You say the warmup/cooldown settings are hard-coded. What are they set to?

I assume they are enabled by default

**Comment by sunfish62** [comment]
OK. I will set aside the Workout settings, since I (still) do not understand their purpose. Thank you for trying to explain them to me; sometimes I can be dense.

Moving on to other areas:

1) I would like to confirm that on the Interval tab, the user can only configure the number and length of intervals and rest. The Warmup and Cooldown settings are hard-coded, and are set to use a tap to continue the sequence. If I have these points correct, I will add them to the wiki description.

2) For the Advanced tab, I have stated on the wiki page that there are 3 workouts included. I have a number of additional questions here, though:
Is there a means by which a user can add workouts? 
Is there an import workout feature? 
If one has a Garmin Connect account and creates a workout there, is there a way to include the workout in RunnerUp?
If a user modifies one of these workouts, is there a way to reset the workout to its original?

**Comment by gerhardol** [comment]
Not the answers you may want to hear...

  > If I have these points correct, I will add them to the wiki description.
Yes

>    Is there a means by which a user can add workouts?
 Settings-Workout-Handle Workouts

<    Is there an import workout feature?
No

 >   If one has a Garmin Connect account and creates a workout there, is there a way to include the workout in RunnerUp?
Garmin sync no longer works (they charge like $!000 for that now)

>    If a user modifies one of these workouts, is there a way to reset the workout to its original?
Dont think so, other than reinstalling

>    Are there ways to add or delete steps in a workout?
Not other creating a new workout

**Comment by sunfish62** [comment]
Thank you for the answers! 

I admit that I did not expect to see adding workouts to be a function of the Settings. Now that I understand this, things are much clearer.

The "No" answers are less critical, given the ability to add workouts. I will write up this information and put it into the wiki. 

Now, since there is a start to a manual, I will close this issue.

**Comment by jonasoreland** [comment]
actually, if you have a garmin account,
you can export it "manually" from that site and import that into runner up
by mailing it with the correct mime-type
and it will be imported to runnerup...
